core/model/modx/modresource.class.php

```diff
@@ -450,6 +450,43 @@ function __construct(xPDO & $xpdo) {
         $this->_cacheKey= "[contextKey]/resources/[id]";
     }
 
+    /**
+     * Compute the "preview URL" for the resource
+     *
+     * @return string
+     */
+    public function getPreviewUrl() {
+        $url = '';
+        if (!$this->get('deleted')) {
```

Comment:
I personally prefer to return values early to make the code more readable. 
```php
if ($this->get('deleted')) {
	return $url;
}
```

Reply:
Same preference, but just copied over existing code. Will update ASAP